### PR TITLE
[TOPI] Specify non-zero absolute tolerance in tests

### PR DESCRIPTION
**Pull request by grwlf**
Refs: #1784

Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

File: python/tvm/testing.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+""" TVM testing utilities """
+import numpy as np
+
+def assert_allclose(actual, desired, rtol=1e-7, atol=1e-7):
```
Comment (masahi) [inline python/tvm/testing.py]: I suggest putting this function under https://github.com/dmlc/tvm/tree/master/topi/python/topi/testing

Since this is only used within topi test.

Comment (grwlf) [reply python/tvm/testing.py]: Agree, including tvm.testing in a patch for TOPI is misleading. But ideally, we need all tests to use non-zero `atol` by default so I still think we need testing utilities on TVM level. Please consider reviewing updated patchset, where all the components are present. tvm.testing is now in TVM patch.

**Comment by masahi** [comment]
can we just use [numpy.allclose](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.allclose.html)?

**Comment by grwlf** [comment]
> can we just use [numpy.allclose](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.allclose.html)?

Do you mean you'd like to replace `np.testing.assert_allclose(..)` with `if not np.allclose(..): raise SomeError(..)` ? I think it is a matter of taste. AFAIK, semantic is the same.

**Comment by masahi** [comment]
I think if there is an equivalent numpy function, we should use it. We don't have to introduce a new function or file. This is such a simple function, I don't expect we would want a customized version or numpy side would break. You can also ask for @tqchen's review.

**Comment by grwlf** [comment]
> I think if there is an equivalent numpy function, we should use it. We don't have to introduce a new function or file. This is such a simple function, I don't expect we would want a customized version or numpy side would break in the future. You can also ask for @tqchen's review.

There is no direct equivalent in numpy. `np.testing.assert_allclose` has default `atol=0` (we investigated this fact and its consequences in #1784). `np.allclose` doesn't report mismatch statistics. Plus see my point about fixing semantics.

File: docker/install/ubuntu_install_python_package.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # install libraries for python package on ubuntu
-pip2 install nose pylint numpy nose-timer cython decorator scipy tornado
-pip3 install nose pylint numpy nose-timer cython decorator scipy tornado typed_ast pytest mypy orderedset antlr4-python3-runtime
+pip2 install nose pylint numpy nose-timer cython decorator scipy tornado psutil
```
Comment (tqchen) [inline docker/install/ubuntu_install_python_package.sh]: seems the changes in docker is introduced un-intentionally?

Comment (grwlf) [reply docker/install/ubuntu_install_python_package.sh]: Fixed. I'll send this patch later.

**Comment by tqchen** [comment]
Thanks @grwlf @masahi this is now merged
